fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/UrlBinary.java

```diff
@@ -89,19 +88,28 @@ public InputStream getContent() {
         }
     }
 
-    @Override
-    public long getContentSize() {
-        final long sizeValue = super.getContentSize();
-        if (sizeValue > -1L) {
-            return sizeValue;
-        }
+    protected long getRemoteContentSize() {
+        final URI resourceUri = getResourceUri();
         try {
-            final String content = IOUtils.toString(getResourceUri().toURL().openStream());
-            setContentSize(content.length());
-            return content.length();
+            final HttpURLConnection httpConn = (HttpURLConnection) resourceUri.toURL().openConnection();
+            httpConn.setRequestMethod("HEAD");
+            httpConn.connect();
+
+            final int status = httpConn.getResponseCode();
+
+            if (status == HttpURLConnection.HTTP_OK) {
+                final String contentLength = httpConn.getHeaderField("Content-Length");
+                if (contentLength != null) {
+                    try {
+                        return Long.parseLong(contentLength);
+                    } catch (final NumberFormatException e) {
+                        LOGGER.debug("Unable to parse Content-Length of remote file {}", resourceUri, e);
```

Comment:
I'm be interested in this being `info` or `warn` versus debug, since it's a pretty serious issue.